Source/Fuse.Nodes/Visual.IsFlat.uno

```diff
@@ -8,86 +9,75 @@ namespace Fuse
 	*/
 	public abstract partial class Visual
 	{
-		internal void InvalidateFlat()
+		// To be overridden to do nothing by flattening nodes (i.e. Viewport)
+		internal virtual void ParentIncrementNonFlat()
 		{
-			if (_isLocalFlatCached || _areChildrenFlatCached)
-			{
-				_isLocalFlatCached = false;
-				_areChildrenFlatCached = false;
-				if (Parent != null)
-					Parent.InvalidateFlat();
-			}
+			IncrementNonFlat();
 		}
-		
-		bool _isLocalFlatCached 
-		{
-			get { return HasBit(FastProperty1.IsLocalFlatCached); }
-			set { SetBit(FastProperty1.IsLocalFlatCached, value); }
-		}
-		bool _isLocalFlat
+
+		// To be overridden to do nothing by flattening nodes (i.e. Viewport)
+		internal virtual void ParentDecrementNonFlat()
 		{
-			get { return HasBit(FastProperty1.IsLocalFlat); }
-			set { SetBit(FastProperty1.IsLocalFlat, value);}
+			DecrementNonFlat();
 		}
 
-		//refers to local transform on the element
-		internal bool IsLocalFlat
+		int _nonFlat = 0;
+		internal void IncrementNonFlat()
 		{
-			get
-			{
-				if (_isLocalFlatCached)
-					return _isLocalFlat;
-					
-				_isLocalFlat = CalcIsLocalFlat();
-				_isLocalFlatCached = true;
-				return _isLocalFlat;
-			}
+			_nonFlat++;
+			if (Parent != null) Parent.ParentIncrementNonFlat();
 		}
-		
-		internal virtual bool CalcIsLocalFlat()
+		internal void DecrementNonFlat()
 		{
-			for (var t = FirstChild<Transform>(); t != null; t = t.NextSibling<Transform>())
-				if (!t.IsFlat) return false;
+			_nonFlat--;
+			if (Parent != null) Parent.ParentDecrementNonFlat();
+		}
 
-			return true;
+		int _localNonFlat = 0;
```

Comment:
It's not clear to me what the difference between _nonFlat and _localNonFlat is (and why _localNonFlat isn't a boolean)...